migrations/Version20230317133142.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20230317133142 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('CREATE TABLE messenger_messages (id BIGINT AUTO_INCREMENT NOT NULL, body LONGTEXT NOT NULL, headers LONGTEXT NOT NULL, queue_name VARCHAR(190) NOT NULL, created_at DATETIME NOT NULL, available_at DATETIME NOT NULL, delivered_at DATETIME DEFAULT NULL, INDEX IDX_75EA56E0FB7336F0 (queue_name), INDEX IDX_75EA56E0E3BD61CE (available_at), INDEX IDX_75EA56E016BA31DB (delivered_at), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+    }
+
+    public function down(Schema $schema): void
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->addSql('DROP TABLE messenger_messages');
+    }
```

Comment:
The `messenger_messages` table is not used by an entity, but is generated on demand by symfony/doctrine-messenger and therefore I'm not sure whether it should be created by a migration or not. 

In https://github.com/itk-dev/naevnssekretariatet we're ignoring this table in migrations (cf. https://github.com/itk-dev/naevnssekretariatet/blob/develop/config/packages/doctrine.yaml#L9-L10). 

One of the projects may be doing this in the wrong way.

Reply:
I'm not sure there is right/wrong here...

My preference is to handle the db schema with migrations. And I prefer that the applications migrations always reflects the db schema needed to run all features of the app. I don't see why this should be limited to Entities? 

I consider doctrines `ignore` feature an escape hatch for situations where `DoctrineMigrations` cannot be made aware of specific tables for whatever reason.

In this specific case the messenger transport integrates with doctrine so migrations are fully aware of the need for the messages table. So why exclude it? I'm aware that the table will be auto-created if missing but this happens by catching a "table missing" exception, then creating the table and re-trying the failed db write. My preference is to avoid depending on exceptions when possible.